### PR TITLE
Make the Buildah Task "tekton chains" friendly

### PR DESCRIPTION
**Pull request by sbose78**
Write out an extra result so that Tekton chains is able to discover what to sign/record.

# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Follows the [authoring recommendations](https://github.com/tektoncd/catalog/blob/main/recommendations.md)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality of task changed or new task added)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [ ] Complies with [Catalog Organization TEP][TEP], see [example]. **Note** [An issue has been filed to automate this validation][validation]
  - [ ] File path follows  `<kind>/<name>/<version>/name.yaml`
  - [ ] Has `README.md` at `<kind>/<name>/<version>/README.md`
  - [ ] Has mandatory `metadata.labels` - `app.kubernetes.io/version` the same as the `<version>` of the resource
  - [ ] Has mandatory `metadata.annotations` `tekton.dev/pipelines.minVersion`
  - [ ] mandatory `spec.description` follows the convention

          ```

          spec:
            description: >-
              one line summary of the resource

              Paragraph(s) to describe the resource.
          ```

_See [the contribution guide](https://github.com/tektoncd/catalog/blob/master/CONTRIBUTING.md)
for more details._

---

[TEP]: https://github.com/tektoncd/community/blob/master/teps/0003-tekton-catalog-organization.md
[example]: https://github.com/tektoncd/catalog/tree/master/task/git-clone/0.1
[validation]:  https://github.com/tektoncd/catalog/issues/413

### CODE REVIEW

**Comment by sbose78** [comment]
_testing this change now_

File: task/buildah/0.3/buildah.yaml [hunk 1]
```diff
@@ -93,7 +95,11 @@ spec:
 
   - name: digest-to-results
     image: $(params.BUILDER_IMAGE)
-    script: cat $(workspaces.source.path)/image-digest | tee /tekton/results/IMAGE_DIGEST
+    script: |
+    
+      cat $(workspaces.source.path)/image-digest | tee /tekton/results/IMAGE_DIGEST
```
Comment (simonbaird) [inline task/buildah/0.3/buildah.yaml:100]: Perhaps change it to tee `$(results.IMAGE_DIGEST.path)` here too for consistency.

File: task/buildah/0.3/buildah.yaml [hunk 2]
```diff
@@ -93,7 +95,11 @@ spec:
 
   - name: digest-to-results
```
Comment (simonbaird) [inline task/buildah/0.3/buildah.yaml:96]: Maybe not important, but you could revise the step name, e.g. "digest-and-url-to-results" or just "results" or "write-results"

File: task/buildah/0.3/buildah.yaml (hunk 1 above)
Comment (vdemeester) [reply task/buildah/0.3/buildah.yaml:100]: @simonbaird definitely 😉

Comment (sbose78) [reply task/buildah/0.3/buildah.yaml]: Done 👍

**Comment by chmouel** [comment]
you need to yamllint your contribution

**Review by vinamra28 — COMMENTED** [review]
@sbose78 thank you for the contribution :+1: 
I wonder if we should bump the version of Task instead of modifying the existing Task? :thinking: 
Reason: existing users would come to know that a version is there which has some new changes that are different from the previous version. WDYT?
/cc @vdemeester @chmouel

**Comment by vdemeester** [comment]
> @sbose78 thank you for the contribution +1 I wonder if we should bump the version of Task instead of modifying the existing Task? thinking Reason: existing users would come to know that a version is there which has some new changes that are different from the previous version. WDYT? /cc @vdemeester @chmouel

So it doesn't really change the behavior, that why I am ok keeping the version. But I don't really have strong opinion there, we could bump it as well.

**Review by vdemeester — APPROVED** [review]

**Comment by vinamra28** [comment]
@sbose78 yamllint seems to be sad :sweat_smile: 
```
task/buildah/0.3/buildah.yaml
  100:1     error    trailing spaces  (trailing-spaces)
  102:1     error    trailing spaces  (trailing-spaces)
```
can you please fix this?

**Comment by vdemeester** [comment]
/lgtm cancel
@sbose78 can you fix the yamllint ?

**Comment by sbose78** [comment]
> can you fix the yamllint ?

On it 🗡️ 

( been really trying hard 😥 )

**Comment by vinamra28** [comment]
@sbose78 thanks for fixing this :bow:  Lastly can you please squash all the commits and then we can merge this?

**Comment by jimmyjones2** [comment]
@sbose78 I've not used tekton chains, but currently IMAGE_URL will be set to a mutable tag, eg. `myregistry.com/org/myimage:latest`. Should it be set to `myregistry.com/org/myimage:latest@sha256:abcdef0123456789` instead?

**Comment by vinamra28** [comment]
@vdemeester should we close this PR as this is now addressed in latest version of buildah?

**Comment by vdemeester** [comment]
@vinamra28 yep.

**Pull request closed without merging**